spec/index.bs

```diff
@@ -4183,41 +4189,113 @@ has a default layout created and used instead.
 
 ### <dfn dictionary>GPUProgrammableStage</dfn> ### {#GPUProgrammableStage}
 
+A {{GPUProgrammableStage}} describes the entry point in the user-provided
+{{GPUShaderModule}} that controls one of the programmable stages of a [=pipeline=].
+
 <script type=idl>
 dictionary GPUProgrammableStage {
     required GPUShaderModule module;
     required USVString entryPoint;
+    record<USVString, double> constants;
 };
 </script>
 
-A {{GPUProgrammableStage}} describes the entry point in the user-provided
-{{GPUShaderModule}} that controls one of the programmable stages of a [=pipeline=].
+<dl dfn-for=GPUProgrammableStage dfn-type=dict-member>
+    : <dfn>constants</dfn>
+    ::
+        Specifies the values of [=pipeline-overridable=] constants which are [=statically accessed=]
+        by the entry point {{GPUProgrammableStage/entryPoint}} in the shader module
+        {{GPUProgrammableStage/module}}.
+
+        Each key-value pair must uniquely identify a [=pipeline-overridable=] constant
+        (by a string representing the numeric ID of the constant, if specified;
```

Comment:
It seems like there is a validation rule missing that you don't use the name if an id was specified in WGSL. Or if that is allowed, that the map does not contain entry for both the name and the id.

Reply:
It is meant to be implied by the "otherwise" (both here and in the WGSL spec changes), but I will add a note.